src/main/java/by/itech/lab/supplier/controller/WarehouseController.java

```diff
@@ -30,4 +37,22 @@
     public WarehouseDto findById(@PathVariable final Long id) {
         return warehouseService.findById(id);
     }
+
+    @PostMapping
+    @ResponseStatus(HttpStatus.CREATED)
+    public WarehouseDto save(@PathVariable(name = "customerId") Long customerId, @Valid @RequestBody WarehouseDto warehouseDto) {
+        return warehouseService.save(customerId, warehouseDto);
+    }
+
+    @PutMapping(URL_ID_PARAMETER)
+    public WarehouseDto updateWarehouse(@PathVariable(name = "customerId") Long customerId, @PathVariable Long id, @Valid @RequestBody WarehouseDto warehouseDto) {
```

Comment:
`@PathVariable(name = "customerId") Long customerId`
`name` not necessary if your variable names as expected

Reply:
Thanks!